addons/mail/static/src/core/common/suggestion_model.js

```diff
@@ -0,0 +1,58 @@
+import { OR, Record } from "@mail/core/common/record";
+
+export class Suggestion extends Record {
+    static id = OR("mention", "channelCommand", "cannedResponse");
+    /** @type {Object.<number, import("models").Suggestion>} */
+    static records = {};
+    /** @returns {import("models").Suggestion} */
+    static get(data) {
+        return super.get(data);
+    }
+    /** @returns {import("models").Suggestion|import("models").Suggestion[]} */
+    static insert(data) {
+        return super.insert(...arguments);
+    }
+    static new() {
+        return super.new(...arguments);
+    }
+
+    mention = Record.one("Mention");
+    channelCommand;
+    cannedResponse = Record.one("CannedResponse");
+
+    get type() {
+        if (this.mention) {
+            return "mention";
+        }
+        if (this.channelCommand) {
+            return "channelCommand";
+        }
+        if (this.cannedResponse) {
+            return "cannedResponse";
+        }
+        throw new Error("Unkown suggestion type");
+    }
+
+    get navigableListInfo() {
+        switch (this.type) {
+            case "mention":
+                return this.mention.navigableListInfo;
+            case "channelCommand":
+                return {
+                    label: this.channelCommand.name,
+                    help: this.channelCommand.help,
+                    optionTemplate: "mail.Composer.suggestionChannelCommand",
+                };
+            case "cannedResponse":
+                return {
+                    cannedResponse: this.cannedResponse,
+                    source: this.cannedResponse.source,
+                    label: this.cannedResponse.substitution,
+                    optionTemplate: "mail.Composer.suggestionCannedResponse",
```

Comment:
alphabetic sort?

Reply:
Currently the suggestions are displayed together with others of the same type. The sorting is already implemented for each single type. The only case where two suggestions of the same type are displayed together is the partner mentions and special mentions. That case gets a special handling for the sorting and then the "group" property handles the visual grouping of the suggestions (special mentions should appear in their own category in order not to be confused with the regular ones).

Reply:
He means to sort the property alphabetically :)
This is a recurring debate: should we sort object property alphabetically or not?

Reply:
Oh, I just kept the original order from `composer.js`:
https://github.com/odoo/odoo/pull/166169/files#diff-6f878c7731d007474053f87b56afc58ee0898b13ba23d1e618e772ad7f9472cfL366